staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion_test.go

```diff
@@ -0,0 +1,542 @@
+			params: params{fixtures.NewReviewWebhookHandler(t, emptyResponseConverter), []string{"v1beta1", "v1beta2"}},
+			checks: checks(expectConversionFailureMessage("empty-response", "expected 1 converted objects")),
+		},
+		{
+			group:  "failure-message",
+			params: params{fixtures.NewReviewWebhookHandler(t, failureResponseConverter("custom webhook conversion error")), []string{"v1beta1", "v1beta2"}},
+			checks: checks(expectConversionFailureMessage("failure-message", "custom webhook conversion error")),
+		},
+	}
+
+	// TODO: Added for integration testing of conversion webhooks, where decode errors due to conversion webhook failures need to be tested.
+	// Maybe we should identify conversion webhook related errors in decoding to avoid triggering this? Or maybe having this special casing
+	// of test cases in production code should be removed?
+	etcd3watcher.SetFatalOnDecodeError(false)
+	defer etcd3watcher.SetFatalOnDecodeError(true)
+
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, apiextensionsfeatures.CustomResourceWebhookConversion, true)()
+	tearDown, config, options, err := fixtures.StartDefaultServer(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	apiExtensionsClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		tearDown()
+		t.Fatal(err)
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		tearDown()
+		t.Fatal(err)
+	}
+	defer tearDown()
+
+	crd := multiVersionFixture.DeepCopy()
+
+	RESTOptionsGetter := serveroptions.NewCRDRESTOptionsGetter(*options.RecommendedOptions.Etcd)
+	restOptions, err := RESTOptionsGetter.GetRESTOptions(schema.GroupResource{Group: crd.Spec.Group, Resource: crd.Spec.Names.Plural})
+	if err != nil {
+		t.Fatal(err)
+	}
+	etcdClient := fixtures.NewCRDTestEtcdClient(t, &restOptions)
+	defer etcdClient.Close()
+
+	etcdObjectReader, err := newEtcdObjectReader(etcdClient, &restOptions, crd)
+	if err != nil {
+		t.Fatal(err)
+	}
+	crdTearDown, crdCtx := newCRDTestContext(t, apiExtensionsClient, dynamicClient, etcdObjectReader, crd)
+	defer crdTearDown()
+
+	// This marker is read only after creation. Since read at a version other than v1beta2 will
+	// force a conversion, we can check if webhook conversion servers are ready by reading this
+	// marker periodically until web observe that the conversion webhook has handled at least 1
+	// request.
+	markerObj, err := crdCtx.versionedClient("marker", "v1beta2").Create(NewVersionedInstance("marker", "marker", "v1beta2"), metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	markerClient := crdCtx.versionedClient("marker", "v1beta1")
+
+	for _, test := range tests {
+		t.Run(test.group, func(t *testing.T) {
+			webhookClientConfig, waitReady, tearDown, err := fixtures.StartConversionWebhookServerAndWaitReady(test.params.handler)
+			if err != nil {
+				t.Fatal(err)
+			}
+			crdCtx.setConversionWebhook(t, []string{"v1beta1", "v1beta2"}, webhookClientConfig)
```

Comment:
v1beta2 is a CustomResource version, but what `crdCtx.setConversionWebhook` and `fixtures.SetWebhookConverter` take as inputs seem to be ConversionReview versions

Reply:
Looks like I can just leave this as the default.